src/app/features/auth/authentication/auth.service.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpClient } from '@angular/common/http';
+import { inject, Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { BehaviorSubject, tap } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { userActions } from '../store/user.action';
+import { AppState } from 'src/app/app.module';
+import { LoginData } from './auth.interface';
+import { API_URL } from '@core/env.token';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  private http = inject(HttpClient);
+  private store = inject<Store<AppState>>(Store);
+  private router = inject(Router);
+  private apiUrl = inject(API_URL);
+  private url = '/login';
+
+  private auth$$ = new BehaviorSubject<{ hasAuth: boolean }>({
+    hasAuth: false,
+  });
+
+  logIn(email: string, password: string) {
+    return this.http
+      .post<LoginData>(this.apiUrl + this.url, {
```

Comment:
Tutaj jak typujesz post jako LoginData to wydaje mi się, że mówisz mu, że w response dostaniesz obiekt typu LoginData który widzę, że ma w sobie m.in email i hasło co nigdy nie przyjdzie nam z serwera

Reply:
W tym przypadku wydaję mi się że LoginData oznacza to co wysyłam w metodzie POST, a nie to co przychodzi z serwera.

Reply:
W końcu w sumie zmieniłem LoginData na interface z polami: `accessToken` oraz user: {type: string, id: number}